src/main/scala/higherkindness/rules_scala/workers/zinc/compile/ZincRunner.scala

```diff
@@ -212,17 +212,25 @@ object ZincRunner extends WorkerMain[ZincRunnerWorkerConfig] {
         isWorker,
       )
 
+    val shouldIncludeSourceRoot = !scalaInstance.actualVersion.startsWith("0.") &&
```

Comment:
You may consider adding a comment here explaining that:
- The `-sourceroot` option only exists in Scala 3
- We only want to add it in Scala 3 because of the TASTy file determinism issue (which also deserves some explanation here)